fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingScreen.kt

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.navigationBarsPadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.statusBarsPadding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.NestedScrollSource
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.PagerState
+import com.google.accompanist.pager.rememberPagerState
+import kotlinx.coroutines.launch
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.compose.PagerIndicator
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private val OnboardingPageTypeList = listOf(
+    JunoOnboardingPageType.DEFAULT_BROWSER,
+    JunoOnboardingPageType.NOTIFICATION_PERMISSION,
+    JunoOnboardingPageType.SYNC_SIGN_IN,
+)
+
+/**
+ * A screen for displaying juno onboarding.
+ *
+ * @param onMakeFirefoxDefaultClick Invoked when the positive button on default browser page is
+ * clicked.
+ * @param onPrivacyPolicyClick Invoked when the privacy policy link text is clicked.
+ * @param onSignInButtonClick Invoked when the positive button on the sign in page is clicked.
+ * @param grantNotificationPermissionButtonClick Invoked when the positive button on notification
+ * page is clicked.
+ * @param onFinish Invoked when the onboarding is completed.
+ */
+@Composable
+fun JunoOnboardingScreen(
+    onMakeFirefoxDefaultClick: () -> Unit,
+    onPrivacyPolicyClick: (String) -> Unit,
+    onSignInButtonClick: () -> Unit,
+    grantNotificationPermissionButtonClick: () -> Unit,
+    onFinish: () -> Unit,
+) {
+    val context = LocalContext.current
+    val coroutineScope = rememberCoroutineScope()
+    val pagerState = rememberPagerState()
+    val isSignedIn: State<Boolean?> = context.components.backgroundServices.syncStore
```

Comment:
```suggestion
    val isSignedIn: State<Boolean?> = context.components.backgroundServices.syncStore
```
There should be an existing `components` getter in a @Composable. https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/samples/compose-browser/src/main/java/org/mozilla/samples/compose/browser/Components.kt#59

Reply:
Good to know, thanks! updated!